packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -187,22 +198,32 @@ export function getLargestContentfulPaint(metric: Metric): ReplayPerformanceEntr
   return getWebVital(metric, 'largest-contentful-paint', node);
 }
 
+function isLayoutShift(entry: PerformanceEntry | LayoutShift): entry is LayoutShift {
+  return (entry as LayoutShift).value !== undefined;
+}
+
 /**
  * Add a CLS event to the replay based on a CLS metric.
  */
 export function getCumulativeLayoutShift(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  const lastEntry = metric.entries[metric.entries.length - 1] as
-    | (PerformanceEntry & { sources?: LayoutShiftAttribution[] })
-    | undefined;
+  const layoutShifts: Attribution[] = [];
   const nodes: Node[] = [];
-  if (lastEntry && lastEntry.sources) {
-    for (const source of lastEntry.sources) {
-      if (source.node) {
-        nodes.push(source.node);
+  for (const entry of metric.entries) {
+    if (isLayoutShift(entry)) {
+      const nodeIds = [];
+      for (const source of entry.sources) {
+        if (source.node) {
+          nodes.push(source.node);
```

Comment:
Nodes is kind of duplicated here since the nodeIds will be in `attributions`, do we use `nodeIds` in the new frontend breadcrumbs?

Reply:
yea we currently need nodeIds in the new frontend to get the selector